sky/provision/vast/utils.py

```diff
@@ -87,21 +87,25 @@ def launch(name: str, instance_type: str, region: str, disk_size: int,
     gpu_name = instance_type.split('-')[1].replace('_', ' ')
     num_gpus = int(instance_type.split('-')[0].replace('x', ''))
 
-    query = ' '.join([
+    query = [
         'chunked=true',
         'georegion=true',
         f'geolocation="{region[-2:]}"',
         f'disk_space>={disk_size}',
         f'num_gpus={num_gpus}',
         f'gpu_name="{gpu_name}"',
         f'cpu_ram>="{cpu_ram}"',
-    ])
+    ]
+    if secure_only:
+        query.append('datacenter=true')
+    query_str = ' '.join(query)
```

Comment:
Nit: I guess we could also do `f'datacenter={secure_only}'`, but I'll leave this up to you.